Mathlib/MeasureTheory/Function/AbsolutelyContinuous.lean

```diff
@@ -0,0 +1,396 @@
+function that maps the finite sequence of the intervals to the total length of the intervals.
+Details:
+1. Technically the filter is on `ℕ × (ℕ → ℝ × ℝ)`. A finite sequence `uIoc (a i) (b i)`, `i < n`
+is represented by any `E : ℕ × (ℕ → ℝ × ℝ)` which satisfies `E.1 = n` and `E.2 i = (a i, b i)` for
+`i < n`. Its total length is `∑ i ∈ Finset.range n, dist (a i) (b i)`.
+2. For a sequence `F : ℕ → ℕ × (ℕ → ℝ × ℝ)`, `F` convergence along `totalLengthFilter` means that
+the total length of `F j`, i.e., `∑ i ∈ Finset.range (F j).1, dist ((F j).2 i).1 ((F j).2 i).2)`,
+tends to `0` as `j` tends to infinity.
+-/
+def totalLengthFilter : Filter (ℕ × (ℕ → ℝ × ℝ)) := Filter.comap
+  (fun E ↦ ∑ i ∈ Finset.range E.1, dist (E.2 i).1 (E.2 i).2) (𝓝[≥] 0)
+
+lemma hasBasis_totalLengthFilter : totalLengthFilter.HasBasis
+    (fun (ε : ℝ) => 0 < ε)
+    (fun (ε : ℝ) => {E | ∑ i ∈ Finset.range E.1, dist (E.2 i).1 (E.2 i).2 < ε}) := by
+  convert Filter.HasBasis.comap _ (nhdsGE_basis _) <;> try infer_instance
+  ext E
+  simp only [mem_setOf_eq, mem_preimage, mem_Ico, iff_and_self]
+  exact fun _ ↦ Finset.sum_nonneg (fun _ _ ↦ dist_nonneg)
+
+/-- The subcollection of all the finite sequences of `uIoc` intervals consisting of
+`uIoc (a i) (b i)`, `i < n` where `a i`, `b i` are all in `uIcc a b` for `i < n` and
+`uIoc (a i) (b i)` are mutually disjoint for `i < n`. Technically the finite sequence
+`uIoc (a i) (b i)`, `i < n` is represented by any `E : ℕ × (ℕ → ℝ × ℝ)` which satisfies
+`E.1 = n` and `E.2 i = (a i, b i)` for `i < n`. -/
+def disjWithin (a b : ℝ) := {E : ℕ × (ℕ → ℝ × ℝ) |
+  (∀ i ∈ Finset.range E.1, (E.2 i).1 ∈ uIcc a b ∧ (E.2 i).2 ∈ uIcc a b) ∧
+  Set.PairwiseDisjoint (Finset.range E.1) (fun i ↦ uIoc (E.2 i).1 (E.2 i).2)}
+
+lemma DisjWithin_comm (a b : ℝ) : disjWithin a b = disjWithin b a := by
+  rw [disjWithin, disjWithin, uIcc_comm]
+
+lemma DisjWithin_mono {a b c d : ℝ} (habcd : uIcc c d ⊆ uIcc a b) : disjWithin c d ⊆ disjWithin a b
+    := by
+  simp +contextual only [disjWithin, Finset.mem_range, setOf_subset_setOf, and_true,
+    and_imp, Prod.forall]
+  exact fun (n I h _ i hi) ↦ ⟨habcd (h i hi).left, habcd (h i hi).right⟩
+
+end AbsolutelyContinuousOnInterval
+
+open AbsolutelyContinuousOnInterval
+
+/-- `AbsolutelyContinuousOnInterval f a b`: A function `f` is *absolutely continuous* on `uIcc a b`
+if the function which applies `f` to every endpoint of the finite sequences of `uIoc` intervals
+tendsto `totalLengthFilter` wrt `totalLengthFilter` restricted to `disjWithin a b`. This is
+equivalent to the traditional `ε`-`δ` definition: for any `ε > 0`, there is `δ > 0` such that for
+any finite disjoint collection of intervals `uIoc (a i) (b i)` for `i < n` where `a i`, `b i` are
+all in `uIcc a b` for `i < n`,  if `∑ i ∈ range n, dist (a i) (b i) < δ`, then
+`∑ i ∈ range n, dist (f (a i)) (f (b i)) < ε`. -/
+def AbsolutelyContinuousOnInterval (f : ℝ → ℝ) (a b : ℝ) :=
```

Comment:
You're only considering functions from R to R, but in fact you could -- and should -- consider functions valued in a general pseudo metric space. `const_mul` and `fun_mul` should be done for functions taking values in a vector space (or rather, you should have `const_smul` and `fun_smul`, and then specializations to `const_mul` and `fun_mul`).

Reply:
For `fun_mul`, I'd rather have a `fun_smul` (with a first function which is scalar valued and the second one which is vector valued), and then deduce the `fun_mul` from the `fun_smul`.